src/wallet/rpcwallet.cpp

```diff
@@ -1877,6 +1877,7 @@ Value walletpassphrase(const Array& params, bool fHelp)
             "walletpassphrase <passphrase> <timeout>\n"
             "Stores the wallet decryption key in memory for <timeout> seconds.");
 
+    pwalletMain->UpdateNullifierNoteMap();
```

Comment:
This doesn't check the return value. (If that's okay, add a comment saying why.)